servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/PublisherFlatMapSingle.java

```diff
@@ -399,27 +364,28 @@ public void onSuccess(@Nullable R result) {
 
             @Override
             public void onError(Throwable t) {
+                Throwable currPendingError = pendingError;
                 if (source.maxDelayedErrors == 0) {
-                    onError0(t, true, true);
+                    if (currPendingError == null &&
+                            pendingErrorUpdater.compareAndSet(FlatMapSubscriber.this, null, t)) {
+                        onError0(t, true);
```

Comment:
Looking at `FlatMapPublisherSubscriber#onError`, it uses `tryEmitItem` in this case instead of `enqueueAndDrain`. Should we align them because the rest of the logic is similar?

Reply:
`tryEmitItem` from the publisher variant attempts to emit and skip the queue if possible (preserves ordering of events relative to that mapped publisher), and enqueues the error if this isn't possible (e.g. lack of demand or elements for that mapped publisher are already queued). since the publisher variant we request demand from each mapped publisher optimistically (to ensure we don't deadlock) this approach avoids queueing events which may require more demand before delivering any errors.

Reply:
Thanks for clarifying